src/responder/kcm/kcmsrv_ops.h

```diff
@@ -29,6 +29,22 @@
 #include "util/sss_iobuf.h"
 #include "responder/kcm/kcmsrv_pvt.h"
 
+/* The initial packet size, which can later be grown up to KCM_PACKET_MAX_SIZE.
+ * The kernel will allocate whole pages even if less is requested, so it
+ * doesn't help to allocate less than one page.
```

Comment:
While it's true that glibc allocator is quite slow, especially when it comes to large chunks, I would better write something more abstract here.
Something about expected average size and performance trade offs.
Take a note that small size actually degrades performance in case of large credentials because realloc is /really/ slow.
But we expect vast majority of creds to be small.

Reply:
Added those information to the comment.

Reply:
I'm stepping on a loose ground here as I'm not really proficient on the topic.
But I'm afraid "kernel pages" / `sysconf(_SC_PAGE_SIZE)` aren't really relevant here.
libc allocator uses `brk()` and `sbrk()` to allocate heap in *virtual* address space.
Then malloc() allocates a buffer within this heap.
A mapping to a real RAM will only happen when program writes to this memory.
Since code doesn't zero entire buffer at allocation anymore, size provided to malloc has no consequences wrt real mem pages allocations.
And I don't even mention that libtalloc wraps malloc with own allocator.

tl,dr: I would keep this really simple, something like:
```
 The initial packet size, which can later be grown up to KCM_PACKET_MAX_SIZE.
Initial size is a trade off that is expected to serve best vast majority of cases (typical credentials size).
```

Reply:
Neither I am an expert on this subject.
The virtual pages are the same and the physical pages, and this is not accident. That's the reason for speaking about the pages (not actually kernel pages, but allocated by the kernel in user space). Then, I cannot say what the `glibc` and `talloc` libraries do.
Anyway, if this drives to confusion, it is better to remove it.